draft.md

```diff
@@ -82,38 +82,68 @@ The following example shows an idempotency key using UUID version 4 scheme:
+        (A) Checksum of the entire request payload.
+        (B) Checksum of selected elements in the request payload.
+        (C) Field value match for each field in the request payload.
+        (D) Field value match for selected elements in the request payload.
+        (E) Request digest/signature.
+
+## Idempotency Enforcement Scenarios
+
+First time Request (Idempotency Key and Idempotency Fingerprint scenarios has not been seen)
+
+    The resource server SHOULD process the request normally and responds with an appropriate response and status code.
+
+Duplicate Request (Idempotency Key and Idempotency Fingerprint scenarios has been seen)
+
+    Replay
+        The request was replayed after the original request completed. The resource server MUST respond with the result of the previously completed operation, success or an    error
+        
+    Concurrent Request
+        The request was replayed before the original request completed. The resoource server MUST respond with a resource conflict error. See #Error scenarions for details.
+       
+## Responsibilities
+
+    Client
+      For the idempotent resource operations, the client MUST present a unique idempotency key in Idempotency-Key request header field.
+      
+    Resource Server
+      Generate Idempotency Fingerprint when required.
+      Check for idempotency under various scenarios including the ones described earlier.
+      Manage the lifecycle of the Idempotency Key.
+      Publish idempotency related specification in relevant documentation.
 
 ## Error Scenarios
 
-If the `Idempotency-Key` request header is missing for a documented idempotent operation requiring this header, the resource MUST reply with an `HTTP` `400` status code with body containing a link pointing to the relevant documentation. Alternately, using the `HTTP` header `Link`, client could be informed about the error too as shown below.
+If the `Idempotency-Key` request header is missing for a documented idempotent operation requiring this header, the resource server MUST reply with an `HTTP` `400` status code with body containing a link pointing to the relevant documentation. Alternately, using the `HTTP` header `Link`, client could be informed about the error too as shown below.
 
     HTTP/1.1 400 Bad Request
     Link: <https://developer.example.com/idempotency>;
       rel="describedby"; type="text/html"
 
-If there is an attempt to reuse an idempotency key with a different request payload, the resource MUST reply with an `HTTP` `422` status code with body containing a link pointing to the relevant documentation. Using the `HTTP` header `Link`, client could be informed about the error as following.
+If there is an attempt to reuse an idempotency key with a different request payload, the resource server MUST reply with an `HTTP` `422` status code with body containing a link pointing to the relevant documentation. Using the `HTTP` header `Link`, client could be informed about the error as following.
```

Comment:
@sdatspun2 @sdatgit  i was trying to change all instances of `resource server` by `resource` and feel like for some of the sentences, the term `resource` wont make sense? what do you think? one example below.

If we use the term resource, it would read as `the resource MUST reply ` (is it meaningful to say `resource` MUST reply or `resource server` MUST reply?)